### DISCUSSION TITLE
이미지 저장 및 주소 반환 방법

### DISCUSSION THREAD

**Discussion by cho-ja-young** [MajorIn-ABO/MajorIn_FE#36] [Polls]
## 프런트가 백엔드에 이미지를 전송하여 백엔드에 저장하고 URL 반환하는 방법

1. 프런트가 html 폼 객체 사용해 이미지 파일 백엔드로 전송
2. POST 요청을 받은 백엔드가 이미지 파일 받아와 지정된 경로에 저장 -> 이때 URL로 변환
3. 저장 경로를 클라이언트에 반환

- 장점:
> 이미지 파일을 백엔드로 직접 전송하기 때문에 이미지 파일이 백엔드 서버로 즉시 전송
> 서버에 이미지가 저장되므로 서버 내에서 이미지에 대한 추가적인 처리를 수행 가능
> 클라우드 서비스에 의존하지 않으므로 서버에서 이미지를 관리하고 보안을 쉽게 유지 가능

- 단점:
> 이미지 파일이 서버에 저장되므로 서버의 용량이나 성능에 영향을 줄 수 있음
> 서버가 다운되면 이미지에 접근할 수 없게 됨

## 프런트가 클라우드 서비스에 이미지를 업로드하고 이미지 URL을 백엔드로 전송하는 방법

1. 프런트엔드에서 이미지 선택
2. AWS SDK 이용해서 해당 파일을 S3에 업로드
3. 업로드된 이미지 URL이 백엔드로 전송
4. 백엔드에서 MYSQL디비에 URL 저장

- 장점:
> 이미지 파일을 클라우드 서비스에 업로드하므로 서버의 부하를 줄일 수 있음
> 클라이언트가 이미지를 업로드한 후 서버에 이미지 URL만 전송하므로 전체적인 트래픽이 감소
> 클라우드 서비스는 보통 높은 가용성과 안정성을 제공하므로 이미지에 대한 안정적인 저장과 제공이 가능

- 단점:
> 이미지 업로드 및 다운로드에 대한 추가 비용이 발생
> 이미지를 클라우드 서비스에 저장하므로 서버 밖에서 이미지에 대한 추가적인 제어가 어려움
> 클라우드 서비스에 의존하므로 서비스가 다운되면 이미지에 접근할 수 없음

**Comment by JangGusWjd** [comment]
2번에 경우에는 프론트에서 서비스 배포에 필요한 서버외에도 이미지 처리를 위한 서버를 새로 생성해야돼서 나는 1번에 투표했어.
그리고 1번 방식으로 할 경우 한 번에 사용자에게 입력을 받아 form data로 백엔드에게 넘겨주면 되는데, 2번 방식의 경우 사용자에게 이미지를 받고 url로 변환하기 위해 서버와 통신하고, 다른 데이터와 함께 벡엔드 서버로 넘겨주기 위해 또다른 서버와 통신을 또 헤야되어서 사용자 입장에서 성능 측면에서도 1번이 더 효율적일 것 같다고 생각해
